File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/ICesiumShadowModeValuePropertyWriter.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 *  
 A writer that can write a value as a shadow mode.
 

 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public interface ICesiumShadowModeValuePropertyWriter extends ICesiumPropertyWriter {
    /**
    *  
    Writes the value as a shadow mode.
    
    

    * @param value The shadow mode.
    */
    void writeShadowMode(@Nonnull CesiumShadowMode value);
}

File: DotNet/CesiumLanguageWriter/Advanced/ICesiumShadowModeValuePropertyWriter.cs
namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// A writer that can write a value as a shadow mode.
    /// </summary>
    public interface ICesiumShadowModeValuePropertyWriter : ICesiumPropertyWriter
    {
        /// <summary>
        /// Writes the value as a shadow mode.
        /// </summary>
        /// <param name="value">The shadow mode.</param>
        void WriteShadowMode(CesiumShadowMode value);
    }
}